pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -138,16 +139,13 @@ func (configgen *ConfigGeneratorImpl) BuildDeltaClusters(proxy *model.Proxy, upd
 	// DeletedClusters contains list of all subset clusters for the deleted DR or updated DR.
 	// When clusters are rebuilt, it rebuilts the subset clusters as well. So, we know what
 	// subset clusters are really needed. So if deleted cluster is not rebuilt, then it is really deleted.
-	var finalDeletedClusters []string
 	builtClusters := sets.New[string]()
 	for _, c := range clusters {
 		builtClusters.Insert(c.Name)
 	}
-	for _, deleted := range deletedClusters {
-		if !builtClusters.Contains(deleted) {
-			finalDeletedClusters = append(finalDeletedClusters, deleted)
-		}
-	}
+	finalDeletedClusters := slices.FilterInPlace(deletedClusters, func(cluster string) bool {
```

Comment:
not sure, is this really more efficient?

Reply:
Sorry. Was on vacation. I did not do any benchmarks but it is supposed to perform better because of less allocations. If you are in doubt, I can quickly setup some benchmark.
cc: @howardjohn

Reply:
Thanks for the reply, I tested it and it is indeed more efficient.